nexus/src/app/mod.rs

```diff
@@ -496,15 +492,36 @@ impl Nexus {
         // not be retried for a while.
         let task_nexus = nexus.clone();
         let task_log = nexus.log.clone();
+        let task_registry = producer_registry.clone();
+        let task_config = config.clone();
         tokio::spawn(async move {
             match task_nexus.wait_for_populate().await {
                 Ok(_) => {
                     info!(
                         task_log,
                         "populate complete; activating background tasks"
                     );
-                    for task in task_nexus.background_tasks.driver.tasks() {
-                        task_nexus.background_tasks.activate(task);
+
+                    let driver = background_tasks_initializer.start(
+                        &task_nexus.background_tasks,
+                        background_ctx,
+                        db_datastore,
+                        task_config.pkg.background_tasks,
+                        rack_id,
+                        task_config.deployment.id,
+                        resolver,
+                        saga_request,
+                        v2p_watcher_channel.clone(),
+                        task_registry,
+                    );
+
+                    if let Err(_) =
+                        task_nexus.background_tasks_driver.set(driver)
+                    {
+                        panic!(
+                            "concurrent initialization of \
+                             background_tasks_driver?!"
+                        )
                     }
```

Comment:
i was briefly unsure about why this doesn't just use `expect(...)` but i realized that it requires the error value to be `Debug`, which i assume this isn't...